docs/udev-configuration.md

```diff
@@ -0,0 +1,148 @@
+```
+To see all the properties of a specific device discovered, you can use `udevadm info`:
+```sh
+udevadm info --attribute-walk --path=$(udevadm info --query=path /sys/devices/pci0000:00/0000:00:1f.3/sound/card0)
+```
+Now, you can see a bunch of attributes you could use to narrow your udev rule. Maybe you decide you want to find all
+sound devices made by the vendor `Great Vendor`. You set the following udev rule under the udev protocol in your
+Configuration:
+```yaml
+spec:
+  protocol:
+    udev:
+      udevRules:
+      -  'SUBSYSTEM=="sound", ATTR{vendor}=="Great Vendor"'
+```
+
+## Using the udev Configuration template
+Instead of having to assemble your own udev Configuration yaml, we have provided a [udev Helm
+template](../deployment/helm/templates/udev.yaml). To add the udev Configuration to your cluster, simple set
+`udev.enabled=true`. Be sure to also **specify one or more udev rules** for the Configuration. Helm will automatically
+apply the udev Configuration yaml for you, and the Akri Agent will advertize discovered leaf devices as resources. By
+default, the udev Configuration does not specify a broker pod or services, so upon discovery, broker pods will not be
+deployed nor will services be created. Later, we will discuss [how to add a custom broker to the
+Configuration](./#adding-a-custom-broker-to-the-configuration).
+```bash
+helm repo add akri-helm-charts https://deislabs.github.io/akri/
+helm install akri akri-helm-charts/akri \
+    --set useLatestContainers=true \
+    --set udev.enabled=true \
+    --set udev.udevRules[0]='SUBSYSTEM=="sound", ATTR{vendor}=="Great Vendor"'
+```
+To discover all sound devices by either Great Vendor or Awesome Vendor, you could add a second udev rule.
+```bash
+helm repo add akri-helm-charts https://deislabs.github.io/akri/
+helm install akri akri-helm-charts/akri \
+    --set useLatestContainers=true \
+    --set udev.enabled=true \
+    --set udev.udevRules[0]='SUBSYSTEM=="sound"\, ATTR{vendor}=="Great Vendor"' \
+    --set udev.udevRules[1]='SUBSYSTEM=="sound"\, ATTR{vendor}=="Awesome Vendor"'
+```
+Akri will now discover these devices and advertize them to the cluster as resources. Each discovered device is
+represented as an Akri Instance. To list them, run `kubectl get akrii`. Note `akrii` is a short name for Akri Instance.
+All the instances will be named in the format `<configuration-name>-<hash>`. You could change the name of the
+Configuration and resultant Instances to be `sound-device` by adding `--set udev.name=sound-devices` to your
+installation command. Now, you can schedule pods that request these Instances as resources. Assuming the Configuration
+name has been set to `sound-devices`, you can request the `sound-device-ha5h00` Instance as a resource by adding the
+following to the PodSpec of your Deployment or Job:
+```yaml
+  resources:
+    limits:
+      akri.sh/sound-device-ha5h00: "1"
+    requests:
```

Comment:
Love this doc. Though wonder (perhaps after this PR), if we should have this documented generically for all protocols.

Reply:
Agreed on the generic and sample, onvif is an exception I think.

Here I was pointing out the fact that you are documenting how to directly take advantage of the generated extended resources (that folks could leverage directly through their pod specs if they want, not through configuration broker specs), rather than just depending on automatic broker deployment. My point was that this would be applicable for other protocols as well, and as such, should be in a protocol agnostic document. Please tell me if you feel otherwise.

Reply:
oh definitely! I can move this out to a different doc